app/controlplane/plugins/core/guac/v1/guac.go

```diff
@@ -0,0 +1,268 @@
+//
+// Copyright 2023 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package guac
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"cloud.google.com/go/storage"
+	"code.cloudfoundry.org/bytefmt"
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	"github.com/chainloop-dev/chainloop/app/controlplane/plugins/sdk/v1"
+	"github.com/go-kratos/kratos/v2/log"
+	"google.golang.org/api/option"
+)
+
+// Integration implements of a FanOut integration
+// See https://github.com/chainloop-dev/chainloop/blob/main/app/controlplane/plugins/README.md for more information
+type Integration struct {
+	*sdk.FanOutIntegration
+}
+
+const providerGCS = "gcs"
+
+// Options provided to the user when registering the integration
+type registrationRequest struct {
+	Provider    string `json:"provider,omitempty" jsonschema:"minLength=1,description=Blob storage provider: default gcs,enum=gcs"`
+	Bucket      string `json:"bucket" jsonschema:"minLength=1,description=Bucket name where to store the artifacts"`
+	Credentials string `json:"credentials" jsonschema:"minLength=2,description=Credentials to access the bucket"`
+}
+
+// No customization options are supported during attachment time
+type attachmentRequest struct{}
+
+// State stored to be retrieved later on during the execution of the actual dispatch
+// NOTE: the credentials are not stored in this state but instead on a secure location, see registrationResponse below
+type registrationState struct {
+	Bucket   string `json:"bucket"`
+	Provider string `json:"provider"`
+}
+
+func New(l log.Logger) (sdk.FanOut, error) {
+	base, err := sdk.NewFanOut(
+		&sdk.NewParams{
+			ID:          "guac",
+			Version:     "1.0",
+			Description: "Export Attestation and SBOMs metadata to a blob storage backend so guacsec/guac can consume it",
```

Comment:
Should we add here that gcs is the only one supported today and ask to create an issue if one of unsupported is needed?

Reply:
I'd not add it here, the limitation is defined in the readme and in the schema.

> ask to create an issue if one of unsupported is needed?

Create an issue where? In Guac?

Reply:
> Create an issue where? In Guac?

In chainloop. First, we have to support it

Reply:
I see, in that case that should go in the readme IMO. I'll update it.